pulser-core/pulser/register/_coordinates.py

```diff
@@ -24,41 +28,44 @@ class CoordsCollection:
         _coords: The coordinates.
     """
 
-    _coords: np.ndarray | list
+    _coords: pm.AbstractArray | list
 
     @property
     def dimensionality(self) -> int:
         """The dimensionality of the coordinates (2 or 3)."""
         return self._sorted_coords.shape[1]
 
     @property
-    def sorted_coords(self) -> np.ndarray:
+    def sorted_coords(self) -> pm.AbstractArray:
         """The sorted coordinates."""
         # Copies to prevent direct access to self._sorted_coords
         return self._sorted_coords.copy()
 
     @cached_property  # Acts as an attribute in a frozen dataclass
-    def _sorted_coords(self) -> np.ndarray:
-        coords = np.array(self._coords, dtype=float)
-        rounded_coords = np.round(coords, decimals=COORD_PRECISION)
-        sorting = self._calc_sorting_order()
-        return cast(np.ndarray, rounded_coords[sorting])
+    def _sorted_coords(self) -> pm.AbstractArray:
+        coords = pm.vstack(cast(Sequence[AbstractArrayLike], self._coords))
+        rounded_coords = pm.round(coords, decimals=COORD_PRECISION)
```

Comment:
Since we are replicating this in `WeightMap`, how about putting these two lines in a `CoordsCollection._rounded_cords` property? You could even cache it

Reply:
Yes, good idea, I can do that.